src/main/java/slimeknights/tconstruct/smeltery/tileentity/AbstractCastingTileEntity.java

```diff
@@ -0,0 +1,234 @@
+package slimeknights.tconstruct.smeltery.tileentity;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.entity.player.PlayerInventory;
+import net.minecraft.fluid.Fluid;
+import net.minecraft.fluid.Fluids;
+import net.minecraft.inventory.ISidedInventory;
+import net.minecraft.inventory.container.Container;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.IRecipeType;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.particles.ParticleTypes;
+import net.minecraft.tileentity.ITickableTileEntity;
+import net.minecraft.tileentity.TileEntityType;
+import net.minecraft.util.Direction;
+import net.minecraft.util.Hand;
+import net.minecraft.util.SoundCategory;
+import net.minecraft.util.SoundEvents;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.BlockRayTraceResult;
+import net.minecraft.util.text.TranslationTextComponent;
+import net.minecraft.world.World;
+import net.minecraft.world.server.ServerWorld;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.common.util.LazyOptional;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.capability.CapabilityFluidHandler;
+import net.minecraftforge.fluids.capability.IFluidHandler;
+import net.minecraftforge.items.ItemHandlerHelper;
+import net.minecraftforge.items.wrapper.SidedInvWrapper;
+import slimeknights.mantle.tileentity.InventoryTileEntity;
+import slimeknights.tconstruct.library.fluid.FluidTankAnimated;
+import slimeknights.tconstruct.library.network.TinkerNetwork;
+import slimeknights.tconstruct.library.recipe.casting.AbstractCastingRecipe;
+import slimeknights.tconstruct.library.smeltery.CastingFluidHandler;
+import slimeknights.tconstruct.smeltery.network.FluidUpdatePacket;
+import slimeknights.tconstruct.smeltery.recipe.ICastingInventory;
+import slimeknights.tconstruct.smeltery.recipe.TileCastingWrapper;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class AbstractCastingTileEntity extends InventoryTileEntity implements ITickableTileEntity, ISidedInventory,
+  FluidUpdatePacket.IFluidPacketReceiver {
+  public static final int INPUT = 0;
+  public static final int OUTPUT = 1;
+  public FluidTankAnimated tank;
+  public LazyOptional<CastingFluidHandler> holder = LazyOptional.of(() -> new CastingFluidHandler(this, tank));
+  private final ICastingInventory crafting;
+  protected int timer;
+  protected AbstractCastingRecipe recipe;
+  protected final IRecipeType<AbstractCastingRecipe> recipeType;
+
+  public AbstractCastingTileEntity(TileEntityType<?> tileEntityTypeIn, IRecipeType<AbstractCastingRecipe> recipeType) {
+    super(tileEntityTypeIn, new TranslationTextComponent("casting"), 2, 1);
```

Comment:
Name seems too short, that will translate it as simply `casting`. While you don't typically have a UI, it will show in the crafting station side inventory, so make it something like `gui.tconstruct.casting`.

Reply:
Gotcha, I just shoved something in there to have a thing.